src/composables/transactionList.ts

```diff
@@ -129,122 +131,46 @@ export function useTransactionList({ store }: IDefaultComposableOptions) {
-          ),
-        ).map((transaction: ITransaction) => ({ ...transaction, pending: true }));
-    } catch (error) {
-      return [];
-    }
-  }
-
-  async function fetchTipWithdrawnTransactions(address: Encoded.AccountAddress, recent: boolean) {
-    try {
-      await getAeSdk();
-      const response = await fetchJson(
-        `${aeActiveNetworkSettings.value.backendUrl}/cache/events/?address=${address}&event=TipWithdrawn${recent ? '&limit=5' : ''}`,
-      );
-      if (response.message) {
-        return [];
-      }
-      // TODO prepare interface for response
-      const tipWithdrawnTransactions: ITransaction[] = (response as any[]).map(({
-        amount,
-        contract,
-        height,
-        data: { tx },
-        ...t
-      }) => ({
-        tx: {
-          ...tx,
-          address,
-          amount,
-          contractId: contract,
-          type: Tag[Tag.ContractCallTx],
-        },
-        ...t,
-        microTime: new Date(t.createdAt).getTime(),
-        blockHeight: height,
-        claim: true,
-      }));
-
-      return tipWithdrawnTransactions;
-    } catch (error) {
-      return [];
-    }
-  }
-
   async function fetchTransactions(
-    limit: number,
-    recent: boolean,
     address: Encoded.AccountAddress,
+    protocol?: Protocol,
   ): Promise<ITransaction[]> {
-    await getAeSdk(); // Ensure the `nodeNetworkId` is established
-
+    await getAeSdk();
     ensureAccountTransactionStateExists(address);
     const transactionState = transactions.value[address];
-
-    if (transactionState.nextPageUrl === null && !recent) {
+    if (!protocol) {
       return [];
     }
+    const recent = transactionState.nextPageUrl === '';
+
+    const adapter = ProtocolAdapterFactory.getAdapter(protocol);
+
+    const response = await adapter.fetchTransactions(
+      address,
+      transactionState.nextPageUrl,
+    );
 
-    const [
-      regularTransactions,
-      pendingTransactions,
-      tipWithdrawnTransactions,
-    ] = await Promise.all([
-      fetchTransactionsFromMiddleware(transactionState, address, recent, limit),
-      fetchPendingTransactions(address),
-      fetchTipWithdrawnTransactions(address, recent),
-    ]);
-
-    const lastRegularTransaction = regularTransactions?.[regularTransactions.length - 1];
+    const pendingTransactions = response.pendingTransactions
+      ?.filter((transaction: ITransaction) => !transactionState.pending[nodeNetworkId.value!]?.find(
+        (tx) => tx?.hash === transaction?.hash,
+      ))
+      .map((transaction) => ({ ...transaction, pending: true })) || [];
+
+    const lastPendingTransaction = pendingTransactions?.[pendingTransactions.length - 1];
     // DEX transaction is represented in 3 objects, only last one should be used
     // this condition checking edge case when not all 3 objects in one chunk
-    if (lastRegularTransaction?.type === AEX9_TRANSFER_EVENT) {
+    if (lastPendingTransaction?.type === AEX9_TRANSFER_EVENT) {
```

Comment:
I think this logic should be moved into the `AeternityAdapter` where we fetch the pending transactions.

Reply:
After having some conversation I see it's not possible for now. Resolving...